python/common/org/python/types/Object.java

```diff
@@ -1018,93 +1018,392 @@ public void __delitem__(org.python.Object index) {
         boolean reflectedChecked = v.type() != w.type()
                 && ((org.python.types.Bool) org.Python.isinstance(w, v.type())).value;
+        boolean v_builtin = isBuiltin(v);
+        boolean w_builtin = isBuiltin(w);
 
+        // Reflective case
         if (reflectedChecked) {
-            result = invokeComparison(w, v, reflOperMethod);
+            if (w_builtin) {
+                result = w.__le__(v);
+            } else {
+                result = invokeComparison(w, v, "__le__");
+            }
+
             if (result != org.python.types.NotImplementedType.NOT_IMPLEMENTED) {
                 return result;
             }
         }
 
-        result = invokeComparison(v, w, operMethod);
+        // Normal case
+        if (v_builtin) {
+            result = v.__ge__(w);
+        } else {
+            result = invokeComparison(v, w, "__ge__");
+        }
+
         if (result != org.python.types.NotImplementedType.NOT_IMPLEMENTED) {
             return result;
         }
 
+        // Now check reflection
         if (!reflectedChecked) {
-            result = invokeComparison(w, v, reflOperMethod);
+            if (w_builtin) {
+                result = w.__le__(v);
+            } else {
+                result = invokeComparison(w, v, "__le__");
+            }
+
             if (result != org.python.types.NotImplementedType.NOT_IMPLEMENTED) {
                 return result;
             }
         }
 
-        if (oper.equals("==")) {
-            return org.python.types.Bool.getBool(v == w);
-        } else if (oper.equals("!=")) {
-            return org.python.types.Bool.getBool(v != w);
+        // Error case
+        if (org.Python.VERSION < 0x03060000) {
+            throw new org.python.exceptions.TypeError(String.format(
+                "unorderable types: %s() %s %s()", v.typeName(), ">=", w.typeName()));
+        } else {
+            throw new org.python.exceptions.TypeError(String.format(
+                "'%s' not supported between instances of '%s' and '%s'", ">=", v.typeName(), w.typeName()));
+        }
+    }
+
+    public static org.python.Object __contains__(org.python.Object v, org.python.Object w) {
+        boolean v_builtin = isBuiltin(v);
+        org.python.Object result = null;
+
+        if (v_builtin) {
+            result = v.__contains__(w);
+        } else {
+            result = invokeComparison(v, w, "__contains__");
+        }
+
+        if (result != org.python.types.NotImplementedType.NOT_IMPLEMENTED) {
+            return result;
+        }
+
+        // Error case
+        if (org.Python.VERSION < 0x03060000) {
+            throw new org.python.exceptions.TypeError(String.format(
+                "unorderable types: %s() %s %s()", v.typeName(), "in", w.typeName()));
+        } else {
+            throw new org.python.exceptions.TypeError(String.format(
+                "'%s' not supported between instances of '%s' and '%s'", "in", v.typeName(), w.typeName()));
+        }
+    }
+
```

Comment:
This (`__contains`) and `__not_contains__` is a bit buggy. There should be no reflective checks (1 in [1,2,3] != [1,2,3] in 1, right?) and those `TypeErrors` are not the right exceptions to throw if there is an unorderable type, etc. I've removed the reflective checks since I'm looking at this piece of code but other than that the code should be unchanged. The problem and tests are tracked here: https://github.com/pybee/voc/pull/877